packages/leancode_contracts/lib/src/types/date_time_offset.dart

```diff
@@ -0,0 +1,80 @@
+/// {@template DateTimeOffset}
+/// A [DateTime] with a fixed offset
+/// {@endtemplate}
+class DateTimeOffset {
+  /// Constructs a [DateTimeOffset] from a utc [DateTime] and a given offset in minutes.
+  /// Assets that the given [DateTime] is UTC and the given offset is between -14h and 14h (inclusive)
+  DateTimeOffset(this.utcDateTime, this.offsetInMinutes)
+      : assert(utcDateTime.isUtc),
+        assert(offsetInMinutes.abs() <= Duration.minutesPerHour * 14);
+
+  /// Deserializes an ISO 8601 string into [DateTimeOffset]
+  factory DateTimeOffset.fromJson(String json) {
+    final match = _parseFormat.firstMatch(json);
+
+    if (match == null) {
+      throw FormatException('Invalid date format', json);
+    }
+
+    var offsetInMinutes = 0;
+    var dateTimeString = json;
+
+    if (match[8] != null) {
+      final tzSign = match[9];
+      if (tzSign != null) {
+        final sign = tzSign == '-' ? -1 : 1;
+        final hourDifference = int.parse(match[10]!);
+        final minuteDifference = match[11] == null ? 0 : int.parse(match[11]!);
+
+        offsetInMinutes =
+            sign * (Duration.minutesPerDay * hourDifference + minuteDifference);
+      }
+
+      dateTimeString = json.substring(0, json.length - match[8]!.length);
+    }
+
+    return DateTimeOffset(DateTime.parse(dateTimeString), offsetInMinutes);
```

Comment:
Here dateTime should be constructed with a forced UTC (`+ 'Z'`). Already fixed it, will push once the review is done